bounswe/bounswe2017group5
# Data templates CRUD UI

## About

* **Related Issues:** #127

We ought to implement the UI for preparation, creation and usage of the data templates. The preparation and creation parts are coarsely done, i.e. looks bad but functionally OK. As for the usage, the user can see the available templates, switch to them, however may not share a post with them.

### Changes

All the changes are on the group details page.

- A modal pops up for template creation.
- One of the three fields can be added to the template: Checkbox, Multiple Selection, and Text Area.
- Labels and legends can be edited through clicking, finalized through clicking away or hitting enter.
- "Create!" button submits the template creation form, into the database.

- Dropdown is now only for selecting the active template, creation is a separate button.
- Dropdown now properly displays the templates of the group.
- Two more buttons are for editing and extending the currently selected template, but not yet functional.

- Sharing posts through templates is not yet available.

## Visuals

See the below comment from @smddzcy.

## Testing

How can this contribution be tested? Use the following structure:

- Assuming your database already has a user and at least one group.
- Click the "Create Template" button.
- Prepare a template through the "+" button and editing labels as described above.
- Submit the template.

- Check back the template on the group details page.
- Check if the template is available on the group details page of another group.

- And more...

---

Closes #127
